### ISSUE TITLE
[BUG] p.y_scale = LogScale() not being updated by 

### ISSUE THREAD

**Issue by gsteele13** [bokeh/bokeh#11983]
### Software versions

Python version      :  3.9.7 (default, Sep 16 2021, 08:50:36) 
IPython version     :  7.29.0
Tornado version     :  6.1
Bokeh version       :  2.4.2
BokehJS static path :  /Users/gsteele/anaconda3/envs/jan2022/lib/python3.9/site-packages/bokeh/server/static
node.js version     :  (not installed)
npm version         :  (not installed)
Operating system    :  macOS-10.16-x86_64-i386-64bit

### Jupyter notebook / Jupyter Lab version

Selected Jupyter core packages... IPython          : 7.29.0 ipykernel        : 6.4.1 ipywidgets       : 7.6.5 jupyter_client   : 7.1.0 jupyter_core     : 4.9.1 jupyter_server   : not installed jupyterlab       : not installed nbclient         : 0.5.3 nbconvert        : 6.3.0 nbformat         : 5.1.3 notebook         : 6.4.6 qtconsole        : 5.1.1 traitlets        : 5.1.1

### Expected behavior

I would expect that if I set `p.y_scale = Logscale()` and then `push_notebook()` that the plot y scale should be come logscale.

### Observed behavior

It has no effect

### Example code

```Python
import numpy as np

from bokeh.models import ColumnDataSource
from bokeh.plotting import figure, show
from bokeh.io import output_notebook, push_notebook
from bokeh.models import LogScale, LinearScale


# This will import the latest compatible bokeh JS from the CDN
output_notebook()

# Generate data
x = np.linspace(0,100,1000)
y = 1/((x-50)**2 + 1) 
y += np.random.normal(size=len(x))*0.01
y += 1/((x-80)**2 + 5)

# Plot it
p = figure(height=200, width=600) 
p.sizing_mode = "scale_width"
p.line(x,y)
# If uncommented, this does work
# p.y_scale = LogScale()
target = show(p, notebook_handle=True)

# But this does nothing
p.y_scale = LogScale()
push_notebook(handle=target)
```

**Comment by bryevdv** [comment]
@gsteele13 swapping out scales on an existing plot is not really supported at present. Best advice would be to create two separate plots up front, and toggle their visibility as appropriate.

**Comment by gsteele13** [comment]
thanks, that was my guess, but wasn't sure. I'll try the two plot trick

**Comment by gsteele13** [comment]
Indeed, this does actually work pretty well, seamless visual transition: 

```
import numpy as np

from bokeh.models import ColumnDataSource
from bokeh.plotting import figure, show
from bokeh.io import output_notebook, push_notebook
from bokeh.models import LogScale, LinearScale

from bokeh.layouts import column
from ipywidgets import interact

p1 = figure(height=200, width=600) 
p1.sizing_mode = "scale_width"
p1.line(x,y)

p2 = figure(height=200, width=600) 
p2.sizing_mode = "scale_width"
p2.line(x,y)
p2.y_scale = LogScale()
p2.visible = False

col = column([p1,p2])
target = show(col, notebook_handle=True)

def update(log=0):
    if log: 
        p1.visible = False
        p2.visible = True
    else:
        p1.visible = True
        p2.visible = False
    push_notebook(handle=target)

interact(update, log=(0,1))
```

**Comment by gsteele13** [comment]
Maybe also: a good idea to lock the two axis scales so that if you zoom and then switch to log, the scale is the same. 

Of course, if you make selections of the cursor tools with one, they will not be selected when you flip to logscale. I guess there is probably a way to lock the cursor selection settings

**Comment by gsteele13** [comment]
of course, one must be careful with negative y-ranges and a logscale...

**Comment by bryevdv** [comment]
LInked selection in this case would be no different than any other case:

https://docs.bokeh.org/en/2.4.2/docs/user_guide/interaction/linking.html#linked-brushing

Linking ranges is probably not advised for the reason you mention, unless perhaps hard positive bounds are configured on the range.
